okio/src/jvmMain/kotlin/okio/internal/ResourceFileSystem.kt

```diff
@@ -42,9 +42,16 @@ import java.net.URL
  */
 @ExperimentalFileSystem
 internal class ResourceFileSystem internal constructor(
-  classLoader: ClassLoader
+  classLoader: ClassLoader,
+  indexEagerly: Boolean,
 ) : FileSystem() {
-  private var roots: List<Pair<FileSystem, Path>> = classLoader.toClasspathRoots()
+  private val roots: List<Pair<FileSystem, Path>> by lazy { classLoader.toClasspathRoots() }
+
+  init {
+    if (indexEagerly) {
+      roots.size
+    }
+  }
```

Comment:
That's clever and weird at the same time...

Reply:
Yeah, I tried to find something more explicit and failed!